File: backend/app/schemas/resume.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class ResumeContent(BaseModel):
    personalInfo: Optional[dict] = None
    education: list = []
    experience: list = []
    skills: list = []
    projects: list = []
    sectionOrder: list = [
        "personalInfo",
        "education",
        "experience",
        "skills",
        "projects",
    ]


class ResumeCreate(BaseModel):
    title: str = "Untitled Resume"
    template_id: str = "modern"
    content: Optional[ResumeContent] = None


class ResumeUpdate(BaseModel):
    title: Optional[str] = None
    template_id: Optional[str] = None
    content: Optional[ResumeContent] = None


class ResumeResponse(BaseModel):
    id: int
    title: str
    template_id: str
    content: Optional[dict] = None
    user_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/resume.ts
export interface PersonalInfo {
  fullName?: string;
  email?: string;
  phone?: string;
  location?: string;
  summary?: string;
  linkedin?: string;
  website?: string;
}

export interface EducationItem {
  id: string;
  school: string;
  degree: string;
  field?: string;
  startDate: string;
  endDate: string;
  description?: string;
}

export interface ExperienceItem {
  id: string;
  company: string;
  role: string;
  location?: string;
  startDate: string;
  endDate: string;
  description?: string;
  current?: boolean;
}

export interface ProjectItem {
  id: string;
  name: string;
  url?: string;
  description?: string;
  tech?: string[];
}

export interface CertificationItem {
  id: string;
  name: string;
  issuer: string;
  date: string;
  url?: string;
}

export type SectionKey =
  | "personalInfo"
  | "education"
  | "experience"
  | "skills"
  | "certifications"
  | "projects";

export interface ResumeContent {
  personalInfo: PersonalInfo | null;
  education: EducationItem[];
  experience: ExperienceItem[];
  skills: string[];
  certifications: CertificationItem[];
  projects: ProjectItem[];
  sectionOrder: SectionKey[];
}

export const DEFAULT_SECTION_ORDER: SectionKey[] = [
  "personalInfo",
  "education",
  "experience",
  "skills",
  "certifications",
  "projects",
];

export const defaultResumeContent: ResumeContent = {
  personalInfo: null,
  education: [],
  experience: [],
  skills: [],
  certifications: [],
  projects: [],
  sectionOrder: [...DEFAULT_SECTION_ORDER],
};

export interface Resume {
  id: number;
  title: string;
  template_id: string;
  content: ResumeContent | null;
  user_id: number;
  created_at: string;
  updated_at: string | null;
}
